test/e2e/hot-reload.test.js

```diff
@@ -71,23 +84,24 @@ describe('hot-reload', () => {
                   delete childEnv.EG_DISABLE_CONFIG_WATCH;
 
                   const modulePath = path.join(__dirname, '../..', 'lib', 'index.js');
-                  childProcess = fork(modulePath, [], {
+                  const options = {
+                    modulePath,
                     cwd: tempPath,
                     env: childEnv
-                  });
-
-                  childProcess.on('error', done);
-
-                  // Not ideal, but we need to make sure the process is running.
-                  setTimeout(() => {
-                    request
-                      .get(`http://localhost:${originalGatewayPort}`)
-                      .end((err, res) => {
-                        should(err).not.be.undefined();
-                        should(res.unauthorized).not.be.undefined();
-                        done();
-                      });
-                  }, GATEWAY_STARTUP_WAIT_TIME);
+                  };
+                  startServer(options)
+                    .then(res => {
+                      request
+                        .get(`http://localhost:${originalGatewayPort}`)
+                        .end((err, res) => {
+                          should(err).not.be.undefined();
```

Comment:
Please use `if (err) { return done(err) }` so Mocha can tell us what the error was.

Reply:
Hi @XVincentX, 
Thanks for the good work on Express Gateway, and thanks for the comment. 
I agree with you, specially the Promise.reject when RegExp fails.
I will do another PR. With a new PR, I'm also happy if you modify any short comings. 
Thanks